### PR TITLE
Update Xilinx NetworkInterface.c

### PR DESCRIPTION
**Pull request by kar-rahul-aws**
Update Xilinx NetworkInterface.c

Description
-----------
Added pxFillinterfaceDescriptor function to accommodate end point changes in IP initialization

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by kar-rahul-aws** [comment]
bot run uncrustify

File: source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c [hunk 1]
```diff
@@ -129,6 +129,25 @@ static BaseType_t prvGMACWaitLS( TickType_t xMaxTimeTicks );
...
 
+/* FreeRTOS+TCP/multi :
+ * Each network device has 3 access functions:
+ * - Initialise the device
+ * - Output a network packet
+ * - Return the PHY Link-Status (LS)
+ * They can be defined as static because the function addresses
+ * addresses will be stored in struct NetworkInterface_t. */
+
+static BaseType_t xNetworkInterfaceInitialise( NetworkInterface_t * pxInterface );
```
Comment (htibosch) [inline source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c:140]: These 3 functions normally have a name that starts like `xCyclone_NetworkInterfaceInitialise()`. Even though these are static functions, I would like to give them unique names.
For instance `xUltraScale_NetworkInterfaceInitialise()`.

( the reason: sometimes I use grep to find a particular NIC function. Then I don't like to see 56 hits that all look similar )

File: source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c [hunk 2]
```diff
@@ -437,10 +457,10 @@ static BaseType_t prvGMACWaitLS( TickType_t xMaxTimeTicks )
 #endif /* ( nicUSE_UNCACHED_MEMORY == 0 ) */
 /*-----------------------------------------------------------*/
 
-BaseType_t xGetPhyLinkStatus( void )
+static BaseType_t xGetPhyLinkStatus( NetworkInterface_t * pxInterface )
```
Comment (htibosch) [inline source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c:460]: See above, change `xGetPhyLinkStatus()` into something like `xUltraScale_GetPhyLinkStatus()`.

File: source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c [hunk 3]
```diff
@@ -453,6 +473,29 @@ BaseType_t xGetPhyLinkStatus( void )
     return xReturn;
 }
 /*-----------------------------------------------------------*/
+NetworkInterface_t * pxFillInterfaceDescriptor( BaseType_t xEMACIndex,
+                                                NetworkInterface_t * pxInterface )
+{
+    static char pcName[ 17 ];
```
Comment (htibosch) [inline source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c:479]: As soon as multiple MAC's are supported, the `pcName[]` should host multiple names., like e.g.:
~~~
static char pcName[ XPAR_XEMACPS_NUM_INSTANCES ][ 17 ];
~~~
Or we could `malloc()` the space needed and pass it to `pxInterface->pcName`.

File: source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c [hunk 4]
```diff
@@ -453,6 +473,29 @@ BaseType_t xGetPhyLinkStatus( void )
...
+
+/* This function pxFillInterfaceDescriptor() adds a network-interface.
+ * Make sure that the object pointed to by 'pxInterface'
+ * is declared static or global, and that it will remain to exist. */
+
+    snprintf( pcName, sizeof( pcName ), "eth%u", ( unsigned ) xEMACIndex );
+
+    memset( pxInterface, '\0', sizeof( *pxInterface ) );
+    pxInterface->pcName = pcName;                    /* Just for logging, debugging. */
+    pxInterface->pvArgument = ( void * ) xEMACIndex; /* Has only meaning for the driver functions. */
```
Comment (htibosch) [inline source/portable/NetworkInterface/xilinx_ultrascale/NetworkInterface.c:489]: Originally, I thought of storing an object here, a pointer to a struct. But most drivers, like this one, store a short number of the EMAC-in-use.

**Review by htibosch — COMMENTED** [review]
That is good work, thank you. I had a few minor comments, please check them.

**Comment by kar-rahul-aws** [comment]
Changes updated in PR#701

**Pull request closed without merging**